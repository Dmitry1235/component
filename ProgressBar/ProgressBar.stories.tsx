import React from 'react';
import { ProgressBar } from './ProgressBar';
import { ProgressBarDocs } from './docs/ProgressBar.docs';

import { withKnobs, boolean, select } from '@storybook/addon-knobs';

export default {
  title: 'Base/ProgressBar',
  decorators: [withKnobs],
  parameters: {
    docs: {
      page: ProgressBarDocs,
    },
  },
};

const DEFAULT_OPTIONS = {
  file: 'file',
  non: '',
};

export const example = () => {
  return (
    <ProgressBar
      isEndAnimation={boolean('isEndAnimation', false)}
      theme={select('theme', DEFAULT_OPTIONS, 'file') as 'file'}
    />
  );
};
