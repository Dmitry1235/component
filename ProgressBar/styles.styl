.ProgressBar {
  background $primaryGrayColor
  overflow hidden
  border-radius 3px
  height 4px

  &__line {
    height 100%
    transform translateX(-100%)
    background #4CACE8
    border-radius 3px
  }

  &__theme {
    &_file {
      & ^[0]__line {
        animation: animFile 1s ease-in-out;
        transform translateX(-10%)
      }
    }
  }

  &__endAnimation {
    & ^[0]__line {
      transition transform 0.3s ease-in-out;
      transform translateX(0%)
    }
  }
}

@keyframes animFile {
  0% {
    transform translateX(-100%)
  }
  50% {
    transform translateX(-50%)
  }
  75% {
    transform translateX(-25%)
  }
  90% {
    transform translateX(-10%)
  }
}