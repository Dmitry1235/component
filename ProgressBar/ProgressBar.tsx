import React from 'react';
import cn from 'classnames/bind';

// Styles
import styles from './styles.styl';
const cx = cn.bind(styles);

export interface ProgressBarProps {
  className?: string;
  theme?: 'file';
  isEndAnimation?: boolean;
}

export const ProgressBar: React.FC<ProgressBarProps> = ({ className, theme, isEndAnimation }) => {
  return (
    <div
      className={cx(
        'ProgressBar',
        { [`ProgressBar__theme_${theme}`]: theme, ProgressBar__endAnimation: isEndAnimation },
        className,
      )}
    >
      <div className={cx('ProgressBar__line')} />
    </div>
  );
};
