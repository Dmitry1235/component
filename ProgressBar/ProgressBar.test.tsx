import React from 'react';
import { ProgressBar } from './ProgressBar';
import renderer from 'react-test-renderer';

describe('ProgressBar:: Snapshots', () => {
  it('Default', () => {
    const component = renderer.create(<ProgressBar isEndAnimation={false} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('ProgressBar:: Render DOM', () => {});
