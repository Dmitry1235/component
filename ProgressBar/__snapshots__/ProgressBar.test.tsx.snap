// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar:: Snapshots Default 1`] = `
<div
  className="ProgressBar"
>
  <div
    className="ProgressBar__line"
  />
</div>
`;
